### ISSUE TITLE
Project Inspect Missing Auth Type Info

### ISSUE THREAD

**Issue by cicoyle** [hashicorp/waypoint#4498]
**Describe the bug**
Upon looking at the `waypoint project inspect` I noticed that we don't output information regarding auth. Not too sure if that was intentional for security purposes or not, but I ran into an err recently and couldn't check if it was set in the `waypoint project inspect`. 

```
$ waypoint project inspect

» Project Info:
              Project Name: my-app
              Applications: my-app
                Workspaces: default
            Remote Enabled: true
               Data Source: Git
                   Git URL: giturl
                   Git Ref: main
  Data Source Poll Enabled: false
   App Status Poll Enabled: false
        File Change Signal: SIGINT
```

I was seeing this err:
<img width="1217" alt="image" src="https://user-images.githubusercontent.com/25733780/217575522-4cf04b81-23ac-415e-b705-303ed7a03f5a.png">

So I ran the next `project apply` to add the private key and auth type. Then ran another project inspect to check if it applied and I couldn't tell.
```
$ waypoint project apply -data-source="git" -git-url="giturl" -git-ref=main -git-private-key-path=/Users/cassiecoyle/Documents/Waypoint/my-priv-key -git-username=user -git-auth-type=ssh my-app
✓ Project "my-app" updated


$ waypoint project inspect

» Project Info:
              Project Name: my-app
              Applications: my-app
                Workspaces: default
            Remote Enabled: true
               Data Source: Git
                   Git URL: giturl
                   Git Ref: main
  Data Source Poll Enabled: false
   App Status Poll Enabled: false
        File Change Signal: SIGINT
```

**Steps to Reproduce**
Create a new project and using a remote runner in EKS. Run `waypoint build`. 

**Expected behavior**
I would've thought I could see the auth type info in the project inspect to confirm that my `project apply` changes worked without re-running `waypoint build` to see if it would err again or work.

**Waypoint Platform Versions**
HCP

**Comment by evanphx** [comment]
You're right, the output needs to include the auth information that's not secret, so that someone can debug what is going to happen.

For secret info, we could output a hash of the value for the user to check.
